core/frontend/src/RealityDataSource.ts

```diff
@@ -208,6 +208,13 @@ export namespace RealityDataSource {
 
     return provider.createRealityDataSource(key, iTwinId);
   }
+
+  /** Returns the source's tileset url if the source is an instance of RealityDataSourceTilesetUrlImpl.
+   * @internal
+   */
+  export function getTilesetUrlFromTilesetUrlImpl(rdSource: RealityDataSource): string | undefined {
```

Comment:
Delete this, add an optional method on the `RealityDataSource` interface, and implement it in `RealityDataSourceTilesetUrlImpl`. That will allow any other implementations of `RealityDataSource` to implement it themselves if they so choose.